### ISSUE TITLE
Sik-Ho Tang | Brief Review -- Rectified Linear Units Improve Restricted Boltzmann Machines.

### ISSUE THREAD

**Issue by NorbertZheng** [NorbertZheng/read-papers#82]
Sik-Ho Tang. [Brief Review — Rectified Linear Units Improve Restricted Boltzmann Machines](https://sh-tsang.medium.com/brief-review-rectified-linear-units-improve-restricted-boltzmann-machines-43724ede3ecf).

**Comment by NorbertZheng** [comment]
## Overview
Rectified Linear Unit (ReLU) Introduced.

Rectified Linear Units Improve Restricted Boltzmann Machines. ReLU, by University of Toronto. 2010 ICML, Over 17000 Citations. Activation Function, Restricted Boltzmann Machine, Image Classification, Face Recognition.

 - Rectified Linear Unit (ReLU) is introduced, which outperforms Sigmoid.
 - This is a paper from Hinton’s research group.

**Comment by NorbertZheng** [comment]
## Rectified Linear Unit (ReLU)
More precisely, Noisy ReLU is proposed to replace the logistic sigmoid function needs to be used many times to get the probabilities required for sampling an integer value correctly:

$$
max(0, x+N(0,\sigma(x))),
$$

where $N(0,\sigma)$ is Gaussian noise with zero mean and variance $\sigma$.

**Comment by NorbertZheng** [comment]
## Image Classification Results
![image](https://user-images.githubusercontent.com/37235023/236660834-8a1f8feb-3861-4c87-887f-dfd81a5c6dee.png)
Network architecture used for the Jittered-Cluttered NORB classification task.

 - Two hidden layers of NReLUs as RBMs, are greedily pretrained. (For RBM, please read [Autoencoder](https://sh-tsang.medium.com/review-autoencoder-reducing-the-dimensionality-of-data-with-neural-networks-data-visualization-fc16446ae32d).)
 - The class label is represented as a $K$-dimensional binary vector with 1-of-$K$ activation, where $K$ is the number of classes.
 - The classifier computes the probability of the $K$ classes from the second layer hidden activities $h_{2}$ using the softmax function.

![image](https://user-images.githubusercontent.com/37235023/236660931-2d2eaec2-a8d6-4cef-9914-40282a9a57b8.png)
Test error rates for classifiers with 4000 hidden units trained on 32×32×2 Jittered-Cluttered NORB images.

 - Pre-training helps improve the performance of both unit types.
 - But NReLUs without pre-training are better than binary units with pre-training.

![image](https://user-images.githubusercontent.com/37235023/236660982-d3c5de4a-2e64-4bb1-aeda-da62829e626d.png)
Test error rates for classifiers with two hidden layers (4000 units in the first, 2000 in the second), trained on 32×32×2 Jittered-Cluttered NORB images.

 - Pre-training both layers gives further improvement for NReLUs but not for binary units.

**Comment by NorbertZheng** [comment]
## Face Recognition Results
![image](https://user-images.githubusercontent.com/37235023/236661009-b34640f9-952f-4733-984b-98de3cc35f5c.png)
Siamese network used for the Labeled Faces in the Wild task.

 - The feature extractor $F_{W}$ contains one hidden layer of NReLUs pre-trained as an RBM. (For RBM, please read [Autoencoder](https://sh-tsang.medium.com/review-autoencoder-reducing-the-dimensionality-of-data-with-neural-networks-data-visualization-fc16446ae32d).)
 - Cosine distance is used to check whether the faces are the same.

![image](https://user-images.githubusercontent.com/37235023/236661069-06280ec0-6821-419c-8c43-68c3a215c542.png)
Accuracy on the LFW task for various models trained on 32×32 colour images
 - Models using NReLUs are more accurate.

**Comment by NorbertZheng** [comment]
This paper and [AlexNet](https://medium.com/coinmonks/paper-review-of-alexnet-caffenet-winner-in-ilsvrc-2012-image-classification-b93598314160?source=post_page---------------------------) are often cited when ReLU is used. Classic!

**Comment by NorbertZheng** [comment]
## Reference
[2010 ICML] [ReLU] [Rectified Linear Units Improve Restricted Boltzmann Machines](https://www.cs.toronto.edu/~fritz/absps/reluICML.pdf).

### Image Classification
1989 … 2010 [ReLU] … 2022 [[ConvNeXt](https://sh-tsang.medium.com/review-convnext-a-convnet-for-the-2020s-53b9ada30ab9)] [[PVTv2](https://sh-tsang.medium.com/review-pvtv2-improved-baselines-with-pyramid-vision-transformer-5cfd354a53bd)]

### Face Recognition
2005 [[Chopra CVPR’05](https://sh-tsang.medium.com/review-learning-a-similarity-metric-discriminatively-with-application-to-face-verification-fd338ae166d6)] 2010 [ReLU] 2014 [[DeepFace](https://sh-tsang.medium.com/973442ad7850)] [[DeepID2](https://sh-tsang.medium.com/review-deep-learning-face-representation-by-joint-identification-verification-deepid2-ee182d4a4a2a)] [[CASIANet](https://sh-tsang.medium.com/c1d976eb0372)] 2015 [[FaceNet](https://sh-tsang.medium.com/review-facenet-a-unified-embedding-for-face-recognition-and-clustering-7b360d2a85e4)] 2016 [[N-pair-mc Loss](https://sh-tsang.medium.com/review-improved-deep-metric-learning-with-multi-class-n-pair-loss-objective-n-pair-mc-loss-2463a7514d8)]
